<a href="https://colab.research.google.com/github/senthilp61/google_colab/blob/main/NumPy.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# The first step of using numpy is to tell python to import it
import numpy as np

**2.1 NumPy Arrays**

NumPy Array


*   An Array is a data structure that stores values of same data type.
*   While python lists can contain values of different data types, arrays in python can only contain values of the same data type.


*   However python lists fail to deliver performance required while computing large sets of numerical data. To address this issue we use NumPy arrays.
*   We can create NumPy arrays by converting a list to an array.




In [ ]:
# define a list of different car companies or string elements
arr_str = ['Mercedes', 'BMW', 'Audi', 'Ferrari', 'Tesla']
print(arr_str)
print(type(arr_str))
# defining a list of number of cylinders in car or numerical elements
arr_numm = [5,4,6,7,3]
print(arr_numm)
print(type(arr_numm))

['Mercedes', 'BMW', 'Audi', 'Ferrari', 'Tesla']
<class 'list'>
[5, 4, 6, 7, 3]
<class 'list'>


In [ ]:
# converting the list arr_str to a NumPy array
np_arr_str = np.array(arr_str)
print("NumPy Array (arr_str): ", np_arr_str)
print(type(np_arr_str))

#converting the list arr_numm to a NumPy array
np_arr_numm = np.array(arr_numm)
print("NumPy Array (arr_numm): ", np_arr_numm)
print(type(np_arr_numm))

NumPy Array (arr_str):  ['Mercedes' 'BMW' 'Audi' 'Ferrari' 'Tesla']
<class 'numpy.ndarray'>
NumPy Array (arr_numm):  [5 4 6 7 3]
<class 'numpy.ndarray'>


## NumPy Matrix


*   A matrix is a two-dimensional data structure where elements are arranged into rows and columns.
*   A matric can be created by using list of lists



In [ ]:
# let's say we have information of different number of cylinders in a car and we want to display them in a matrix format
matrix = np.array([[1,2,1],[4,5,9],[1,8,9]])
print(matrix)

[[1 2 1]
 [4 5 9]
 [1 8 9]]


In [ ]:
print("Data type of matrix: ", type(matrix))

Data type of matrix:  <class 'numpy.ndarray'>




*   We see that all the NumPy objects have data type of ndarray



# 2.2 NumPy Functions
There are different ways to create NumPy arrays using the functions available in NumPy library

Using np.arange() function

In [ ]:
arr2 = np.arange(start=0, stop=10) # 10 will be excluded from the output
print(arr2)

# or

arr2 = np.arange(0,10)
print(arr2)

[0 1 2 3 4 5 6 7 8 9]
[0 1 2 3 4 5 6 7 8 9]


In [ ]:
# adding a step size of 5 to create an array
arr3 = np.arange(start=0, stop=20, step=5)
print(arr3)

# or

arr3 = np.arange(0,20,5)
arr3

[ 0  5 10 15]


array([ 0,  5, 10, 15])

### Using np.linspace() function


*   The np.linspace() function returns numbers which are evenly distributed with respect to interval. Here the start and stop both are included
*   It has the following parameters:


*   **start**: start of interval range. By default start = 0
*   **stop**: end of interval range
*   **num**: No. of samples to generate. By default num = 50






In [ ]:
matrix2 = np.linspace(0,5) # by default 50 evenly spaced values will be generated between 0 and 5
matrix2

array([0.        , 0.10204082, 0.20408163, 0.30612245, 0.40816327,
       0.51020408, 0.6122449 , 0.71428571, 0.81632653, 0.91836735,
       1.02040816, 1.12244898, 1.2244898 , 1.32653061, 1.42857143,
       1.53061224, 1.63265306, 1.73469388, 1.83673469, 1.93877551,
       2.04081633, 2.14285714, 2.24489796, 2.34693878, 2.44897959,
       2.55102041, 2.65306122, 2.75510204, 2.85714286, 2.95918367,
       3.06122449, 3.16326531, 3.26530612, 3.36734694, 3.46938776,
       3.57142857, 3.67346939, 3.7755102 , 3.87755102, 3.97959184,
       4.08163265, 4.18367347, 4.28571429, 4.3877551 , 4.48979592,
       4.59183673, 4.69387755, 4.79591837, 4.89795918, 5.        ])

**How are these values getting generated?**
The step size or the difference between each element will be decided by the following formula:

**(stop - start)/(total elements - 1)**

so, in this case: (5-0)/49 = 0.10204082

The first value will be 0.10204082, the second value will be 0.10204082 + 0.10204082, and so on

In [ ]:
# generating 10 evenly spaced values between 10 and 20
matrix3 = np.linspace(10,20,10)
matrix3

array([10.        , 11.11111111, 12.22222222, 13.33333333, 14.44444444,
       15.55555556, 16.66666667, 17.77777778, 18.88888889, 20.        ])

**Similarly we can create matrices using the functions available in NumPy library**

**Using np.zeros()**


*   The np.zeros() is a function for creating a matrix and performing matrix operations in NumPy.
*   It returns the matrix filled with zeros of the given shape.


*   It has the following parameters:
*   **shape**: Number of rows and columns in the output matrix.


*   **dtype**: data type of the elements in the matrix, by default the value is set to **float**.







In [ ]:
matrix4 = np.zeros([3,5])
matrix4

array([[0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0.]])

**Using np.ones()**

In [ ]:
matrix5 = np.ones([3,5])
matrix5

array([[1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1.]])

**Using np.eye()**

In [ ]:
matrix6 = np.eye(5)
matrix6

array([[1., 0., 0., 0., 0.],
       [0., 1., 0., 0., 0.],
       [0., 0., 1., 0., 0.],
       [0., 0., 0., 1., 0.],
       [0., 0., 0., 0., 1.]])

**We can also convert a one dimension array to a matrix. This can be done by using the np.reshape() function.**

In [ ]:
# defining an array with values 0 to 9
arr4 = np.arange(0,10)
arr4

array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])

In [ ]:
# reshaping arr4 to a 2 x 5 matrix
arr4_reshaped = arr4.reshape((2,5))
arr4_reshaped

array([[0, 1, 2, 3, 4],
       [5, 6, 7, 8, 9]])

**NumPy can also perform a large number of different mathematical operations and it provides different functions to do so.**

NumPy provides:


1.   Trignometric functions
2.   Exponents and Logarithmic functions
3.   Functions for arithmetic operations between arrays and matrices





**Trignometric functions**

In [ ]:
print("Sine Function:", np.sin(4))
print("Cosine Function:", np.cos(4))
print("Tan Functions:", np.tan(4))

Sine Function: -0.7568024953079282
Cosine Function: -0.6536436208636119
Tan Functions: 1.1578212823495775


**Exponents and Logarithmic functions**
*   Exponents






In [ ]:
np.exp(2)

7.38905609893065

In [ ]:
arr5 = np.array([2,4,6])
np.exp(arr5)

array([  7.3890561 ,  54.59815003, 403.42879349])



*   Logarithms



In [ ]:
# by default NumPy takes the base of log as e
np.log(2)

0.6931471805599453

In [ ]:
np.log(arr5)

array([0.69314718, 1.38629436, 1.79175947])

In [ ]:
## log with base 10
np.log10(8)

0.9030899869919435

**Arithmetic Operations on arrays**

In [ ]:
# arithmetic on lists

l1 = [1,2,3]
l2 = [4,5,6]
print(l1+l2)
# this doesnot behave as you would expect

[1, 2, 3, 4, 5, 6]


In [ ]:
# We can +-*/ arrays together

# defining two arrays
arr7 = np.arange(1,6)
print('arr7:',arr7)

arr8 = np.arange(3,8)
print('arr8:', arr8)

arr7: [1 2 3 4 5]
arr8: [3 4 5 6 7]


In [ ]:
print("Addition", arr7+arr8)

Addition [ 4  6  8 10 12]


In [ ]:
print("Subtraction: ",arr8-arr7)

Subtraction:  [2 2 2 2 2]


In [ ]:
print("Multiplication: ", arr7*arr8)

Multiplication:  [ 3  8 15 24 35]


In [ ]:
print("Division:", arr7/arr8)

Division: [0.33333333 0.5        0.6        0.66666667 0.71428571]


In [ ]:
print("Inverse:", 1/arr7)

Inverse: [1.         0.5        0.33333333 0.25       0.2       ]


In [ ]:
print("Powers", arr7**arr8) # in python, powers are acheived using **, NOT ^!!! ^ does something completely different

Powers [    1    16   243  4096 78125]


**Operations on Matrices**

In [ ]:
matrix7 = np.arange(1,10).reshape(3,3)
print(matrix7)

[[1 2 3]
 [4 5 6]
 [7 8 9]]


In [ ]:
matrix8 = np.eye(3)
print(matrix8)

[[1. 0. 0.]
 [0. 1. 0.]
 [0. 0. 1.]]


In [ ]:
print("Addition: \n", matrix7+matrix8)
print("Subtraction: \n", matrix7-matrix8)
print("Multiplication: \n", matrix7*matrix8)
print("Division: \n", matrix7/matrix8)

Addition: 
 [[ 2.  2.  3.]
 [ 4.  6.  6.]
 [ 7.  8. 10.]]
Subtraction: 
 [[0. 2. 3.]
 [4. 4. 6.]
 [7. 8. 8.]]
Multiplication: 
 [[1. 0. 0.]
 [0. 5. 0.]
 [0. 0. 9.]]
Division: 
 [[ 1. inf inf]
 [inf  5. inf]
 [inf inf  9.]]


<ipython-input-62-ad0cf6eca0c3>:4: RuntimeWarning: divide by zero encountered in divide
  print("Division: \n", matrix7/matrix8)


**Linear algebra matrix multiplication**

In [ ]:
matrix9 = np.arange(1,10).reshape((3,3))
print("First matrix: \n", matrix9)

matrix10 = np.arange(11,20).reshape((3,3))
print("Second matrix: \n", matrix10)
print('')
# taking linear algebra matrix multiplication (some may have heard this called the dot product)
print("Multiplication: \n", matrix9 @ matrix10)

First matrix: 
 [[1 2 3]
 [4 5 6]
 [7 8 9]]
Second matrix: 
 [[11 12 13]
 [14 15 16]
 [17 18 19]]

Multiplication: 
 [[ 90  96 102]
 [216 231 246]
 [342 366 390]]


**Transpose of a matrix**

In [ ]:
print(matrix9)

[[1 2 3]
 [4 5 6]
 [7 8 9]]


In [ ]:
# taking transpose of matrix
np.transpose(matrix9)


array([[1, 4, 7],
       [2, 5, 8],
       [3, 6, 9]])

In [ ]:
# another way of taking a transpose
matrix9.T

array([[1, 4, 7],
       [2, 5, 8],
       [3, 6, 9]])

**Function to find minimum and maximum values**

In [ ]:
print(matrix9)

[[1 2 3]
 [4 5 6]
 [7 8 9]]


In [ ]:
print("Minimum value: ", np.min(matrix9))

Minimum value:  1


In [ ]:
print("Maximum value: ", np.max(matrix9))

Maximum value:  9


**Function to generate random samples**

**Using np.random.rand function**


*   The np.random.rand returns a random NumPy array whose element(s) are drawn randomly from the normal distribution over [0,1). (including 0 but excluding 1).
*   **Syntax** - np.random.rand(d0,d1)
*   d0,d1 - It represents the dimension of the required array given as int, where d1 is optional.





In [ ]:
# Generating random values in an array
rand_mat = np.random.rand(5)
print(rand_mat)

[0.01470471 0.54360533 0.27762507 0.2783502  0.91313231]


In [ ]:
# * Generating random values in a matrix
rand_mat = np.random.rand(5,5) # uniform random variable
print(rand_mat)

[[0.6888297  0.13215444 0.51892071 0.014402   0.59773829]
 [0.59581776 0.3879284  0.74342829 0.81694896 0.30050831]
 [0.22601392 0.84289432 0.87461127 0.5201508  0.20782241]
 [0.12567584 0.38724884 0.53435292 0.91730622 0.07149387]
 [0.33908439 0.14052953 0.22587479 0.48511662 0.92393258]]


**Using np.random.randn function**


*   The np.random.randn returns a random numpy array whose samples are drawn randomly from the standard normal distribution (Mean as 0 and standard deviation as1)
*   **Syntax -** np.random.randn(d0,d1)
*   d0,d1 - It represents the dimension of the output, where d1 is optional.





In [ ]:
# Generating random values in an array
rand_mat2 = np.random.randn(5)
print(rand_mat2)

[-0.9470089  -1.70582687  0.79071789 -2.42633832  0.29849094]


In [ ]:
# Generating random values in a matrix
rand_mat2 = np.random.randn(5,5)
print(rand_mat2)

[[-0.88153595 -2.1051843  -3.0737576  -1.08017793  0.72098952]
 [ 0.96313417  0.27205262  0.35305252  1.05392033 -0.45286813]
 [-3.04757748  0.07708321  0.34519914 -1.99853019 -0.25301357]
 [-0.88953719 -0.41293492 -1.11896195 -0.77769948 -1.0876253 ]
 [ 1.01163151 -0.48477245 -1.95339361 -0.76858754  0.89309513]]


In [ ]:
# Let's check the mean and standard deviation of rand_mat2
print("Mean:", np.mean(rand_mat2))
print("Standard Deviation:", np.std(rand_mat2))

Mean: -0.5878399779753591
Standard Deviation: 1.167592559476986




*   We observe mean is very close to 0 and Standard deviation is very close to 1



**Using np.random.randint function**


*   The np.random.randint returns a random numpy array whose element(s) are drawn randomly from low(inclusive) to the high(exclusive) range.
*   **Syntax -** np.random.randint(low,high,size)
*   low - It represents the lowest inclusive bound of the distribution from where the samples can be drawn.
*   high - It represents the upper exclusive bound of the distribution from where the sample can be drawn.
*   size - It represents the shape of the output







In [ ]:
#Generating random values in an array
rand_mat3 = np.random.randint(1,5,10)
print(rand_mat3)

[4 1 4 2 4 2 3 1 4 2]


In [ ]:
# Generating random values in a matrix
rand_mat3 = np.random.randint(1,10,[5,5])
print(rand_mat3)

[[5 9 9 4 9]
 [6 1 3 1 4]
 [9 5 1 5 7]
 [5 7 5 4 4]
 [3 3 8 6 2]]


## 2.3 Accessing the entries of a Numpy Array

In [ ]:
#let's generate an array with 10 random values
rand_arr = np.random.randn(10)
print(rand_arr)

[ 2.46074685  0.53332206  0.05471114  0.25256692 -0.09816916  1.02062909
 -2.00377139  1.85055658 -0.52874951  1.43956171]




*   Accessing one element from an array



In [ ]:
# accessing the 6th entry of rand_arr
rand_arr[5]

1.020629086322696



*   Accessing multiple elements from an array



In [ ]:
# We can access multiple entries at once using
print(rand_arr[4:9])

[-0.09816916  1.02062909 -2.00377139  1.85055658 -0.52874951]


In [ ]:
# We can also access multiple non-consecutive entries using np.arange
print("Index of values to access: ", np.arange(3,10,3))
print(rand_arr[np.arange(3,10,3)])

Index of values to access:  [3 6 9]
[ 0.25256692 -2.00377139  1.43956171]


**Accessing arrays using logical operations**

In [ ]:
print(rand_arr)

[ 2.46074685  0.53332206  0.05471114  0.25256692 -0.09816916  1.02062909
 -2.00377139  1.85055658 -0.52874951  1.43956171]


In [ ]:
rand_arr>0

array([ True,  True,  True,  True, False,  True, False,  True, False,
        True])

In [ ]:
# accessing all the values of rand_arr which are greater than 0
print("Values greated than 0: ",rand_arr[rand_arr>0])

# accessing all the values of rand_arr which are less than 0
print("Values less than 0: ", rand_arr[rand_arr<0])

Values greated than 0:  [2.46074685 0.53332206 0.05471114 0.25256692 1.02062909 1.85055658
 1.43956171]
Values less than 0:  [-0.09816916 -2.00377139 -0.52874951]


**Accessing the entries of a matrix**

In [ ]:
# let's generate an array with 10 random values
rand_mat = np.random.randn(5,5)
print(rand_mat)

[[-1.30756665  0.7464859   1.41019045 -0.61038992 -0.08331082]
 [ 0.92763063  0.07961133 -0.2473379   1.07305938  1.28269395]
 [ 0.24964236 -0.09316885 -0.14075728  1.40281252  0.77745414]
 [ 0.46254025  0.40471811  0.53196612 -0.80131424 -0.01527189]
 [-0.77270196 -0.99519871 -2.70085895  1.45649888 -0.18620778]]


In [ ]:
# accessing the second row of the rand_mat
rand_mat[1]

array([ 0.92763063,  0.07961133, -0.2473379 ,  1.07305938,  1.28269395])

In [ ]:
# accessing third element of the second row
print(rand_mat[1][2])

#or

print(rand_mat[1,2])

-0.2473378978243566
-0.2473378978243566


In [ ]:
# accessing first two rows with second and third column
print(rand_mat[0:2,1:3])

[[ 0.7464859   1.41019045]
 [ 0.07961133 -0.2473379 ]]


**Accessing matrices using logical operations**

In [ ]:
print(rand_mat)

[[-1.30756665  0.7464859   1.41019045 -0.61038992 -0.08331082]
 [ 0.92763063  0.07961133 -0.2473379   1.07305938  1.28269395]
 [ 0.24964236 -0.09316885 -0.14075728  1.40281252  0.77745414]
 [ 0.46254025  0.40471811  0.53196612 -0.80131424 -0.01527189]
 [-0.77270196 -0.99519871 -2.70085895  1.45649888 -0.18620778]]


In [ ]:
# accessing all the values of rand_mat which are greater than 0
print("Values greater than 0: \n", rand_mat[rand_mat>0])

# accessing all the values of rand_mat which are lesser than 0
print("Values lesser than 0: \n", rand_mat[rand_mat<0])

Values greater than 0: 
 [0.7464859  1.41019045 0.92763063 0.07961133 1.07305938 1.28269395
 0.24964236 1.40281252 0.77745414 0.46254025 0.40471811 0.53196612
 1.45649888]
Values lesser than 0: 
 [-1.30756665 -0.61038992 -0.08331082 -0.2473379  -0.09316885 -0.14075728
 -0.80131424 -0.01527189 -0.77270196 -0.99519871 -2.70085895 -0.18620778]


**Modifying the entries of an Array**

In [ ]:
print(rand_mat)

[[-1.30756665  0.7464859   1.41019045 -0.61038992 -0.08331082]
 [ 0.92763063  0.07961133 -0.2473379   1.07305938  1.28269395]
 [ 0.24964236 -0.09316885 -0.14075728  1.40281252  0.77745414]
 [ 0.46254025  0.40471811  0.53196612 -0.80131424 -0.01527189]
 [-0.77270196 -0.99519871 -2.70085895  1.45649888 -0.18620778]]


In [ ]:
print(rand_arr)

[ 2.46074685  0.53332206  0.05471114  0.25256692 -0.09816916  1.02062909
 -2.00377139  1.85055658 -0.52874951  1.43956171]


In [ ]:
# let's change some values in an array!
# changing the values of index value 3 and index value 4 to 5

rand_arr[3:5] = 5
print(rand_arr)

[ 2.46074685  0.53332206  0.05471114  5.          5.          1.02062909
 -2.00377139  1.85055658 -0.52874951  1.43956171]


In [ ]:
# changing the values of index value 0 and index value 1 to 2 and 3 respectively
rand_arr[0:2] = [2,3]
print(rand_arr)

[ 2.          3.          0.05471114  5.          5.          1.02062909
 -2.00377139  1.85055658 -0.52874951  1.43956171]


In [ ]:
# modify entries using logical references
rand_arr[rand_arr >0] = 65
rand_arr

array([65.        , 65.        , 65.        , 65.        , 65.        ,
       65.        , -2.00377139, 65.        , -0.52874951, 65.        ])

**Modifying the entries of a Matrix**

In [ ]:
print(rand_mat3)

[[5 9 9 4 9]
 [6 1 3 1 4]
 [9 5 1 5 7]
 [5 7 5 4 4]
 [3 3 8 6 2]]


In [ ]:
# changing the values of the 4th and 5th element of the second and third rows of the matrix to 0
print("Matrix before modification: \n", rand_mat3)
rand_mat3[1:3,3:5] = 0
print("Matrix after modification: \n", rand_mat3)

Matrix before modification: 
 [[5 9 9 4 9]
 [6 1 3 1 4]
 [9 5 1 5 7]
 [5 7 5 4 4]
 [3 3 8 6 2]]
Matrix after modification: 
 [[5 9 9 4 9]
 [6 1 3 0 0]
 [9 5 1 0 0]
 [5 7 5 4 4]
 [3 3 8 6 2]]


In [ ]:
# extracting the first 2 rows and first 3 columns from the matrix
sub_mat = rand_mat[0:2,0:3]
print(sub_mat)

[[-1.30756665  0.7464859   1.41019045]
 [ 0.92763063  0.07961133 -0.2473379 ]]


In [ ]:
# changing all the values of the extracted matrix to 3
sub_mat[:] = 3
print(sub_mat)

[[3. 3. 3.]
 [3. 3. 3.]]


In [ ]:
#what happend to rand_mat when we change sub_mat
print(rand_mat)

[[ 3.          3.          3.         -0.61038992 -0.08331082]
 [ 3.          3.          3.          1.07305938  1.28269395]
 [ 0.24964236 -0.09316885 -0.14075728  1.40281252  0.77745414]
 [ 0.46254025  0.40471811  0.53196612 -0.80131424 -0.01527189]
 [-0.77270196 -0.99519871 -2.70085895  1.45649888 -0.18620778]]


In [ ]:
# to prevent this behavior we need to use the .copy() method when we assign sub_mat
# this behavior is the source of MANY errors for early python users!!!

rand_mat = np.random.randn(5,5)
print(rand_mat)
print(" ")
rand_mat_int = np.random.randint(1,10,10).reshape((2,5))
print(rand_mat_int)
sub_mat = rand_mat[0:2,0:3].copy()
print(sub_mat)
sub_mat[:] = 3
print(sub_mat)
print(rand_mat)

[[ 1.67401283 -0.91908533 -0.09670656 -0.50196762 -0.35249735]
 [-0.38676123  0.04278323 -1.20743354 -0.78990649 -1.32815847]
 [-0.41830577  0.16783965  0.46408052 -1.03244674  1.47594856]
 [-2.03849691 -0.52562626 -0.99849588  1.61491161 -0.66329306]
 [ 2.02123271 -1.51973168  0.49559223  0.03200765  0.60023688]]
 
[[6 4 7 1 5]
 [5 8 6 9 9]]
[[ 1.67401283 -0.91908533 -0.09670656]
 [-0.38676123  0.04278323 -1.20743354]]
[[3. 3. 3.]
 [3. 3. 3.]]
[[ 1.67401283 -0.91908533 -0.09670656 -0.50196762 -0.35249735]
 [-0.38676123  0.04278323 -1.20743354 -0.78990649 -1.32815847]
 [-0.41830577  0.16783965  0.46408052 -1.03244674  1.47594856]
 [-2.03849691 -0.52562626 -0.99849588  1.61491161 -0.66329306]
 [ 2.02123271 -1.51973168  0.49559223  0.03200765  0.60023688]]


## 2.4 Saving and Loading a NumPy array

### Lets save some NumPy objects on the disk for use later!

In [ ]:
from google.colab import drive
drive.mount("/content/drive")

Mounted at /content/drive


In [ ]:
# creating a random matrices
randint_matrix1 = np.random.randint(1,10,10).reshape((2,5))
print(randint_matrix1)
print(" ")
randint_matrix2 = np.random.randint(10,20,10).reshape((2,5))
print(randint_matrix2)

[[9 9 8 6 5]
 [3 6 3 9 7]]
 
[[15 13 12 13 17]
 [10 13 13 10 10]]


**Using np.save() function**

In [ ]:
np.save("/content/drive/MyDrive/MDS_2/saved_file_name", randint_matrix1)

**Using np.savez() function**

In [ ]:
np.savez("/content/drive/MyDrive/MDS_2/multiple_files", randint_matrix1=randint_matrix1, randint_matrix2=randint_matrix2)



*   The files will be saved in the directory where the Jupyter Notebook is located.
*   With np.save() function, we can save an array/matrix to a NumPy .npy format
*   np.savez() function has an advantage over np.save() function because with np.savez(), we can store several arrays/matrices into a single file in uncompressed .npz format.





In [ ]:
# now let's load it
loaded_arr = np.load("/content/drive/MyDrive/MDS_2/saved_file_name.npy")
loaded_multi = np.load("/content/drive/MyDrive/MDS_2/multiple_files.npz")

print(loaded_arr)
print("")
print(loaded_multi)

[[9 9 8 6 5]
 [3 6 3 9 7]]



In [ ]:
print("1st Matrix: \n", loaded_multi['randint_matrix1'])
print("2nd Matrix: \n", loaded_multi['randint_matrix2'])

new_matrix = loaded_multi['randint_matrix1']
print("New Matrix: \n", new_matrix)

1st Matrix: 
 [[9 9 8 6 5]
 [3 6 3 9 7]]
2nd Matrix: 
 [[15 13 12 13 17]
 [10 13 13 10 10]]
New Matrix: 
 [[9 9 8 6 5]
 [3 6 3 9 7]]


In [ ]:
# We can also save/load text files...but only single variables
np.savetxt("/content/drive/MyDrive/MDS_2/text_file_name.txt", randint_matrix1, delimiter=',')
rand_mat_txt = np.loadtxt("/content/drive/MyDrive/MDS_2/text_file_name.txt", delimiter=',')
print(randint_matrix1)
print('')
print(rand_mat_txt)

[[9 9 8 6 5]
 [3 6 3 9 7]]

[[9. 9. 8. 6. 5.]
 [3. 6. 3. 9. 7.]]


In [ ]:
test_array = [10, 11.5, 12, 13.5, 14,15]
test_array[2:3]

[12]

In [ ]:
import numpy as np
list_l =[[12,34,55,],[66,45,77],[45,77,88]]
np.array(list_l)

array([[12, 34, 55],
       [66, 45, 77],
       [45, 77, 88]])

In [ ]:
import pandas as pd
score = [10, 15, 20, 25]
pd.Series(data=score, index = ['a','b','c','d'])

a    10
b    15
c    20
d    25
dtype: int64

In [ ]:
np.arange(0,22,6)

array([ 0,  6, 12, 18])

In [ ]:
demo_array = np.arange(10,21)
subset_demo_array = demo_array[0:7]
subset_demo_array[:]= 101
subset_demo_array

array([101, 101, 101, 101, 101, 101, 101])

In [ ]:
flowers = pd.Series([2, 3, 5, 4], index=['lily', 'rose', 'daisy', 'lotus'])
flowers['daisy']

5

In [ ]:
s1 = pd.Series(['a', 'b'])
s2 = pd.Series(['c', 'd'])

In [ ]:
pd.concat([s1, s2])


0    a
1    b
0    c
1    d
dtype: object

In [ ]:
football = [1,2,3,4,5]
np.array(football)

array([1, 2, 3, 4, 5])